includes/civicrm.basepage.php

```diff
@@ -81,6 +81,11 @@ public function register_hooks() {
     // regardless of URL, load page template
     add_filter( 'template_include', array( $this, 'basepage_template' ), 999 );
 
+    if (!isset( $_GET['q']) && !empty($_GET['amp;q'])) {
```

Comment:
In fact I hit a situation where $_GET['q'] was 'civicrm/' & $_GET['amp;q'] is 'civicrm/payment/ipn' - it seems that we should always prefer 'amp;q' if set from my experience